x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrStatsAction.java

```diff
@@ -0,0 +1,156 @@
+import org.elasticsearch.action.support.tasks.BaseTasksResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.tasks.Task;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+public class CcrStatsAction extends Action<CcrStatsAction.TasksResponse> {
+
+    public static final String NAME = "cluster:monitor/ccr/stats";
+
+    public static final CcrStatsAction INSTANCE = new CcrStatsAction();
+
+    private CcrStatsAction() {
+        super(NAME);
+    }
+
+    @Override
+    public TasksResponse newResponse() {
+        return null;
+    }
+
+    public static class TasksResponse extends BaseTasksResponse implements ToXContentObject {
+
+        private final List<TransportCcrStatsAction.TaskResponse> taskResponses;
+
+        public TasksResponse() {
+            this(Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
+        }
+
+        TasksResponse(
+                final List<TaskOperationFailure> taskFailures,
+                final List<? extends FailedNodeException> nodeFailures,
+                final List<TransportCcrStatsAction.TaskResponse> taskResponses) {
+            super(taskFailures, nodeFailures);
+            this.taskResponses = taskResponses;
+        }
+
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            final Map<String, Map<Integer, TransportCcrStatsAction.TaskResponse>> taskResponsesByIndex = new TreeMap<>();
+            for (final TransportCcrStatsAction.TaskResponse taskResponse : taskResponses) {
+                taskResponsesByIndex.computeIfAbsent(
+                        taskResponse.followerShardId().getIndexName(),
+                        k -> new TreeMap<>()).put(taskResponse.followerShardId().getId(), taskResponse);
+            }
+            builder.startObject();
+            {
+                for (final Map.Entry<String, Map<Integer, TransportCcrStatsAction.TaskResponse>> index : taskResponsesByIndex.entrySet()) {
+                    builder.startObject(index.getKey());
+                    {
+                        for (final Map.Entry<Integer, TransportCcrStatsAction.TaskResponse> shard : index.getValue().entrySet()) {
+                            builder.startObject(Integer.toString(shard.getKey()));
+                            {
+                                final ShardFollowNodeTask.Status status = shard.getValue().status();
```

Comment:
why can't we use the status toXContent?

Reply:
Because there are slight differences in what we present on the endpoint and the status x-content (which is there to enable round-tripping status to/from x-content). For example: the endpoint doesn't present the index metadata version, and presents successful/failed fetches and bulk operations and total/successful.

Reply:
re index metadata version - why not present everything we have? the purpose of status object is to report information about the running task, which I think is identical with the purpose of this api?

> presents successful/failed fetches and bulk operations as total/successful.

We can change the status class to track these if you feel it's more natural. It's not a part of the task run time behavior so nothing stops us to align these.

Reply:
> re index metadata version - why not present everything we have? 

This does not seem useful from an end-user perspective but we can add in the interest of simplification.

> We can change the status class to track these if you feel it's more natural. It's not a part of the task run time behavior so nothing stops us to align these.

Tracking successful/failed feels more natural internally, presenting total/successful feels more natural but we can present successful/failed in the interest of simplification.

I pushed 55300a8fb20dcb74e7333652869751438dc0cf31.